### ISSUE TITLE
Parsing error: Invalid regular expression flag on \u (unicode) and \s (sticky) modifiers

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#7983]
**Tell us about your environment**

* **ESLint Version:**

3.14.0

* **Node Version:**

7.4.0

* **npm Version:**

4.0.5

**What parser (default, Babel-ESLint, etc.) are you using?**

Default.

**Please show your full configuration:**

```yml
extends: eslint:recommended
env:
  node: true
  browser: true
rules:
  block-scoped-var: 2
  callback-return: 2
  complexity: [2, 13]
  curly: [2, multi-or-nest, consistent]
  dot-location: [2, property]
  dot-notation: 2
  indent: [2, 2, SwitchCase: 1]
  linebreak-style: [2, unix]
  new-cap: 2
  no-console: [2, allow: [warn, error]]
  no-else-return: 2
  no-eq-null: 2
  no-fallthrough: 2
  no-invalid-this: 2
  no-return-assign: 2
  no-shadow: 1
  no-trailing-spaces: 2
  no-use-before-define: [2, nofunc]
  quotes: [2, single, avoid-escape]
  semi: [2, always]
  strict: [2, global]
  valid-jsdoc: [2, requireReturn: false]
  no-control-regex: 0
```

**What did you do? Please include the actual source code causing the issue.**

```js
var URL = /test/u;
```

**What did you expect to happen?**

ESLint should lint the code.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/gajus/Documents/dev/gajus/ajv/lib/compile/formats.js
  12:12  error  Parsing error: Invalid regular expression flag
```

`\u` and `\s` modifiers have been introduced to ES6 (see https://mathiasbynens.be/notes/es6-unicode-regex).

**Comment by not-an-aardvark** [comment]
What happens if you add this to your config?

```yml
parserOptions:
  ecmaVersion: 6
```

`u` and `y` flags are ES6 syntax, so you need to configure the parser for ES6 in order to successfully parse them.

(Also, there is no `s` flag at the moment -- sticky regexes have the `y` flag.)

**Comment by gajus** [comment]
You are right – sorry for a false issue.

Adding `ecmaVersion` configuration fixes the problem.
